lib/class/art.class.php

```diff
@@ -1103,8 +1108,11 @@ public function gather($options = array(), $limit = 0)
 
             return array();
         }
-        $config  = AmpConfig::get('art_order');
-        $methods = get_class_methods('Art');
+        if ($limit == 0) {
+            $limit   = (is_null(AmpConfig::get('art_search_limit'))) ? ART_SEARCH_LIMIT : AmpConfig::get('art_search_limit') ;
```

Comment:
@wagnered Sorry to bug you again, but using `ART_SEARCH_LIMIT` won't work as it's a class constant and not a global one. :(

self::ART_SEARCH_LIMIT or static::ART_SEARCH_LIMIT will do the trick (I personally prefer `static::` because of late static binding https://www.php.net/manual/de/language.oop5.late-static-bindings.php)